substrate/frame/bounties/src/lib.rs

```diff
@@ -808,6 +824,52 @@ pub mod pallet {
 			Self::deposit_event(Event::<T, I>::BountyExtended { index: bounty_id });
 			Ok(())
 		}
+
+		/// Approve bountry and propose a curator simultaneously.
+		/// This call is a shortcut to calling `approve_bounty` and `propose_curator` separately.
```

Comment:
This call seems like is an example of #5958.

Reply:
But just looking at the diff, and knowing little about this pallet, I disagree with this statement. 

If this call was purely an equivalent of `batch(approve, propose)`, then:

1. The implementation should be merely a `approve().and(propose)`. Perhaps this would be less efficient, but it should be functionally the same.
2. There would be no need to introduce a new `BountyStatus` variant. 

So perhaps this PR is actually doing more than just introducing a shorthand for combining two existing atomic operations?

Reply:
The issue is ( https://github.com/paritytech/polkadot-sdk/pull/5961#discussion_r1808423634 ) that after bounty is approved some time must pass until bounty reaches `Funded` status because bounty fundings happen in the beginning of every block.

The only way for this to be atomic is if with approval we try to fund the bounty right away, but there may not be enough money in the treasury and such action will revert.

I'm open to simpler solutions which may keep the flow backwards compatible with previous version.

Reply:
Thanks for explaining! The current solution is good, I was mainly interested in the relationship between this and #5958.